common/src/main/java/org/apache/comet/parquet/ParquetColumnSpec.java

```diff
@@ -19,28 +19,45 @@
 
 package org.apache.comet.parquet;
 
+import java.util.Map;
+
 public class ParquetColumnSpec {
 
+  private final int fieldId;
   private final String[] path;
   private final String physicalType;
   private final int typeLength;
   private final boolean isRepeated;
   private final int maxDefinitionLevel;
   private final int maxRepetitionLevel;
 
+  // Logical type info
+  private String logicalTypeName;
+  private Map<String, String> logicalTypeParams;
+
   public ParquetColumnSpec(
+      int fieldId,
       String[] path,
       String physicalType,
       int typeLength,
       boolean isRepeated,
       int maxDefinitionLevel,
-      int maxRepetitionLevel) {
+      int maxRepetitionLevel,
+      String logicalTypeName,
```

Comment:
Wondering if we can directly pass `LogicalTypeAnnotation` rather than passing a `String` and reconstructing...

Reply:
I don't think we can pass _any_ Parquet class between Comet <-> Iceberg without running into shading issues?